platform-hub-api/app/controllers/announcement_templates_controller.rb

```diff
@@ -0,0 +1,88 @@
+class AnnouncementTemplatesController < ApiJsonController
+
+  before_action :find_announcement_template, only: [ :show, :update, :destroy ]
+
+  authorize_resource
+
+  # GET /announcement_templates
+  def index
+    @announcement_templates = AnnouncementTemplate.order(:shortname)
+
+    render json: @announcement_templates
+  end
+
+  # GET /announcement_templates/1
+  def show
+    render json: @announcement_template
+  end
+
+  # POST /announcement_templates
+  def create
+    @announcement_template = AnnouncementTemplate.new(announcement_template_params)
+
+    if @announcement_template.save
+      AuditService.log(
+        context: audit_context,
+        action: 'create',
+        auditable: @announcement_template
+      )
+
+      render json: @announcement_template, status: :created
+    else
+      render_model_errors @announcement_template.errors
+    end
+  end
+
+  # PATCH/PUT /announcement_templates/1
+  def update
+    if @announcement_template.update(announcement_template_params)
```

Comment:
We could use that approach, but so far for `update` "endpoints we've been using the check-boolean-else-return-the-errors-object" (so the `else` branch of this would return the errors with an appropriate status code). What do you think?